test/conference_ticket.js

```diff
@@ -0,0 +1,139 @@
+  beforeEach(async function(){
+    ct = await ConferenceTicket.new('')
+    this.accounts = accounts
+    this.createConference = async ({
+      name = '',
+      deposit = toWei('0.02', "ether"),
+      limitOfParticipants = 20,
+      coolingPeriod = 0,
+      ownerAddress = accounts[0],
+      clearFee = 10,
+      ticketAddress = ct.address,
+      gasPrice = toWei('1', 'gwei')
+    }) => {
+      let conference = await Conference.new(
+        name,
+        deposit,
+        limitOfParticipants,
+        coolingPeriod,
+        ownerAddress,
+        clearFee,
+        ticketAddress
+        , {gasPrice:gasPrice}
+      );
+      await ct.setConferenceAddress(conference.address);
+      return conference;
+    }
+  })
+
+  describe('on new', function(){
+    it('metadata uri is empty', async function(){
+      await ct.baseTokenURI().should.eventually.eq('')
+    })
+    it('conference address is empty', async function(){
+      await ct.conferenceAddress().should.eventually.eq(emptyAddress)
+    })
+  })
+
+  describe('can set baseTokenURI', function(){
+    it('success if admin', async function(){
+      await ct.baseTokenURI().should.eventually.eq('')
+      await ct.setBaseTokenURI(newBaseTokenURI, { from: accounts[0]})
+      await ct.baseTokenURI().should.eventually.eq(newBaseTokenURI)
+    })
+
+    it('fail if not admin', async function(){
+      await ct.baseTokenURI().should.eventually.eq('')
+      await ct.setBaseTokenURI(newConferenceAddress, { from: accounts[1]}).should.be.rejected
+      await ct.baseTokenURI().should.eventually.eq('')
+    })
+  })
+
+  describe('can set conferenceAddress', function(){
+    it('success if admin', async function(){
+      await ct.conferenceAddress().should.eventually.eq(emptyAddress)
+      await ct.setConferenceAddress(newConferenceAddress, { from: accounts[0]})
+      await ct.conferenceAddress().should.eventually.eq(newConferenceAddress)
+    })
+
+    it('fail if not admin', async function(){
+      await ct.conferenceAddress().should.eventually.eq(emptyAddress)
+      await ct.setConferenceAddress(newConferenceAddress, { from: accounts[1]}).should.be.rejected
+      await ct.conferenceAddress().should.eventually.eq(emptyAddress)
+    })
+  })
+
+  it('can set admins', async() => {
+    await ct.setConferenceAddress(newConferenceAddress, { from: accounts[1]}).should.be.rejected
+    await ct.conferenceAddress().should.eventually.eq(emptyAddress)
+    await ct.grant([accounts[1]], { from: accounts[0]})
+    await ct.setConferenceAddress(newConferenceAddress, { from: accounts[1]}).should.be.fulfilled
+    await ct.conferenceAddress().should.eventually.eq(newConferenceAddress)
+  })
+
+  describe('on mint', function(){
+    it('success if admin', async function(){
+      let tokenId = 123
+      await ct.tokenURI(tokenId).should.be.rejected
+      await ct.mint(accounts[1], tokenId).should.be.fulfilled
+      await ct.tokenURI(tokenId).should.eventually.eq(tokenId.toString())
```

Comment:
Shouldn't this be `{baseUri}/{registrantID}`?

Reply:
If the `baseTokenURI` has not been set it should just return the `tokenId`.

https://github.com/wearekickback/contracts/blob/1bf8b61c76fd8cbb66c41c58db5369cf997b28f1/test/conference_ticket.js#L101-L104